python/cudf/cudf/core/column/string.py

```diff
@@ -1109,3 +1506,32 @@ def _string_column_binop(lhs, rhs, op):
     _ = libcudf.binops.apply_op(lhs=lhs, rhs=rhs, out=out, op=op)
     nvtx_range_pop()
     return out
+
+
+def _get_cols_list(others):
+    from cudf.core import Series, Index
+    from cudf.core.column import as_column
+
+    if (
+        is_list_like(others)
+        and others
+        and (
+            is_list_like(others[0])
+            or isinstance(others[0], (Series, Index, pd.Series, pd.Index))
+        )
```

Comment:
I think we can remove this check entirely  and then below wrap the list comprehension with `as_column` into a try except that raises the Exception in the `else` clause.

Reply:
Connected offline and we need to handle a list-like of string scalars here as well so this isn't straightforward to do. Resolving and approving.